### DISCUSSION TITLE
How to create a versionedclientset?

### DISCUSSION THREAD

**Discussion by mensylisir** [prometheus-operator/prometheus-operator#4379] [Q&A]
Now, I have a Kubernetes.Clientset, It is generated through Kubernetes Bearer Token and the token has cluster-admin permission. It can normally operate all kubernetes non CRD resources, then, I create a versioned.Clientset ,as follow:
```
func getMonitorClient(clusterName string) (*versioned.Clientset, error) {
	client, err := getBaseParam(clusterName) # Kubernetes.Clientset, It can get all resouce from kubernetes 
	if err != nil {
		return nil, err
	}
	monitoringClient := versioned.New(client.DiscoveryClient.RESTClient())
	return monitoringClient, nil
}

func GetPrometheusRule(custername, namespace, prometheusrule_name) (*monitoringV1.PrometheusRule, error) {
	client, err := getMonitorClient(custername)
	if err != nil {
		return nil, err
	}
	pr, err := client.MonitoringV1().PrometheusRules(namespace).Get(context.TODO(), prometheusrule_name, metav1.GetOptions{})
	if err != nil {
		return nil, err
	}
	return pr, nil
}
```

The monitoringClient cannot get prometheusrule, err is not null
```
{"msg":"v1.GetOptions is not suitable for converting to \"meta.k8s.io/v1\" in scheme \"pkg/runtime/scheme.go:101\""}
```

**Comment by shenshouer** [comment]
I met the some problem!

**Reply by noovertime7** [reply]
I met the some problem too!

**Comment by noovertime7** [comment]
Using the client created by monitoringclient.New(client.DiscoveryClient.RESTClient()) will report a notRegiste error when calling, but I use monitoringclient.NewForConfig(config) and it is normal. Haven't found the reason yet   [example](https://github.com/rook/nfs/blob/33558fba3d75ea7c8c657d85ddc9fb9de7c542c2/pkg/operator/k8sutil/prometheus.go)

**Reply by noovertime7** [reply]
@shenshouer
